### PR TITLE
 Support linting code with Credo

### PR DESCRIPTION
**Pull request by korobkov**
https://hex.pm/packages/credo

### CODE REVIEW

**Comment by chrismccord** [comment]
Can you amend the PR to exclude the `credo` dep? Thanks!

**Comment by korobkov** [comment]
Yes, for sure…
Done :)

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

File: lib/phoenix/socket/message.ex [hunk 1]
```diff
@@ -30,7 +30,7 @@ defmodule Phoenix.Socket.Message do
       }
     rescue
       err in [KeyError] ->
-        raise Phoenix.Socket.InvalidMessageError, message: "missing key #{inspect err.key}"
+        reraise Phoenix.Socket.InvalidMessageError, message: "missing key #{inspect err.key}"
```
Comment (josevalim) [inline lib/phoenix/socket/message.ex]: This is wrong, you are not passing the stacktrace (and the previous code is fine). This is a false positive from credo.

File: lib/phoenix/transports/long_poll.ex [hunk 2]
```diff
@@ -124,7 +124,7 @@ defmodule Phoenix.Transports.LongPoll do
   defp decode({:ok, body, conn}, serializer) do
     assign(conn, :message, serializer.decode!(body, []))
   rescue
-    Phoenix.Socket.InvalidMessageError -> raise Plug.Parsers.ParseError
+    Phoenix.Socket.InvalidMessageError -> reraise Plug.Parsers.ParseError
```
Comment (josevalim) [inline lib/phoenix/transports/long_poll.ex]: This is also wrong.

**Pull request merged**